checker.cs

```diff
@@ -3,36 +3,168 @@
 
 class Checker
 {
-    static bool batteryIsOk(float temperature, float soc, float chargeRate) {
-        if(temperature < 0 || temperature > 45) {
-            Console.WriteLine("Temperature is out of range!");
+   /// <summary>
+    /// Batteries the is ok.
+    /// </summary>
+    /// <param name="temperature">The temperature.</param>
+    /// <param name="soc">The soc.</param>
+    /// <param name="chargeRate">The charge rate.</param>
+    /// <returns></returns>
+    static bool batteryIsOk(float temperature, float soc, float chargeRate)
+    {
+        bool temperatureConstrainCheck = checkTemperature(temperature);
+        bool chargeStateConstrainCheck = checkStateOfCharge(soc);
+        bool chargeRateConstrainCheck = checkChargeRate(chargeRate);
+        return (temperatureConstrainCheck && chargeStateConstrainCheck && chargeRateConstrainCheck);
+    }
+
+    /// <summary>
+    /// Checks the temperature.
+    /// </summary>
+    /// <param name="temperature">The temperature.</param>
+    /// <returns></returns>
+    static bool checkTemperature(float temperature)
```

Comment:
avoid camel casing for function names in C#

Reply:
Sir, how can we name the function without camel casing. For instance, instead of CheckTemperature().., can we name the function as check_temperature(). Is there any best practice to name the function?